clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2682,19 +2704,53 @@ void SYCLIntegrationHeader::emit(raw_ostream &O) {
   }
   O << "};\n\n";
 
+  O << "// array representing kernel parameters used in all kernels defined in "
+       "the\n";
+  O << "// corresponding source\n";
+  O << "static constexpr\n";
+  O << "const bool kernel_param_used[] = {\n";
+  O << "  //PARAM_USED_TABLE_BEGIN \n";
+  for (unsigned I = 0; I < KernelDescs.size(); I++) {
+    auto &K = KernelDescs[I];
+    O << "  //--- " << K.Name << "\n";
+    O << "  ";
+    for (unsigned I = 0; I < K.Params.size(); I++) {
+      auto &P = K.Params[I];
+      for (unsigned J = 0; J < P.Count; J++) {
+        O << "1";
+        O << ", ";
```

Comment:
Is the extra "," ok here? Other places had comma even after the last element in an array.

Reply:
Yep, those aren't a problem.